pkg/controller/master/virtualmachine/register.go

```diff
@@ -51,5 +52,18 @@ func Register(ctx context.Context, management *config.Management, options config
 	}
 	virtualMachineInstanceClient.OnChange(ctx, vmControllerSetDefaultManagementNetworkMac, vmNetworkCtl.SetDefaultNetworkMacAddress)
 
+	// register the vm volume status controller, which tries to update LH Volume status to PVC and VM
+	var volumeClient = management.LonghornFactory.Longhorn().V1beta1().Volume()
+	var vmVolumeStatusController = &VMVolumeStatusController{
+		vmClient:  vmClient,
+		vmCache:   vmCache,
+		pvcClient: pvcClient,
+		pvcCache:  pvcCache,
+	}
+	volumeClient.OnChange(ctx, vmControllerSetAnnotationsVolumeStatusName, vmVolumeStatusController.OnLHVolumeChange)
+	volumeClient.OnRemove(ctx, vmControllerSetAnnotationsVolumeStatusName, vmVolumeStatusController.OnLHVolumeRemove)
```

Comment:
True. A better solution rely on LH. Here we can only try to cover some special and important cases.

Reply:
@guangbochen :
**Conclusion: PVC/PV is not holding on "Pending", "Pending" is indeed the init/default status, will move to "Bund" quickly.**


The status `Pending` only exists in any PVC for a very short time, it will be updated to `Bund` quickly, no matter its size, it seems as long as LH creates the volume object, it will update PVC to `Bund`.

**The WebUI is also checking the "Bund" status, then display as "Ready"** , that is why it also changes quickly.

```
k get pvc v-2g -o JSON
    "status": {
        "phase": "Pending"
    }
    
        "status": {
        "accessModes": [
            "ReadWriteMany"
        ],
        "capacity": {
            "storage": "2Gi"
        },
        "phase": "Bound"
    }

k get pvc vol-300g -o JSON
    "status": {
        "accessModes": [
            "ReadWriteMany"
        ],
        "capacity": {
            "storage": "300Gi"
        },
        "phase": "Bound"
    }
```

    
![image](https://user-images.githubusercontent.com/31133476/157340212-977c66f1-fb03-4abb-bdc9-f28687503707.png)

k8s source code 

```
	// VolumePending means the Volume is pending and cannot be attached to the node yet.
	VolumePending VolumePhase = "Pending"
	// VolumeBound means the Volume is bound and can be attach to the node.
	VolumeBound VolumePhase = "Bound"

type PersistentVolumePhase string

const (
	// used for PersistentVolumes that are not available
	VolumePending PersistentVolumePhase = "Pending"
	// used for PersistentVolumes that are not yet bound
	// Available volumes are held by the binder and matched to PersistentVolumeClaims
	VolumeAvailable PersistentVolumePhase = "Available"
	// used for PersistentVolumes that are bound
	VolumeBound PersistentVolumePhase = "Bound"
```

vendor/k8s.io/kubernetes/pkg/apis/core/v1/defaults.go
```
func SetDefaults_PersistentVolume(obj *v1.PersistentVolume) {
	if obj.Status.Phase == "" {
		obj.Status.Phase = v1.VolumePending
	}
	if obj.Spec.PersistentVolumeReclaimPolicy == "" {
		obj.Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimRetain
	}
	if obj.Spec.VolumeMode == nil {
		obj.Spec.VolumeMode = new(v1.PersistentVolumeMode)
		*obj.Spec.VolumeMode = v1.PersistentVolumeFilesystem
	}
}
func SetDefaults_PersistentVolumeClaim(obj *v1.PersistentVolumeClaim) {
	if obj.Status.Phase == "" {
		obj.Status.Phase = v1.ClaimPending
	}
}
```
PV, PVC status are same>
![image](https://user-images.githubusercontent.com/31133476/157341639-1d1d5dc3-f436-444f-a53b-58a7afa5497e.png)

Reply:
Are u using the `master-head` Harvester version with embedded LH `v1.2.4-rc1`, this did not happen to me as creating an `out-of-range` disk will get it stuck on the pending status and its related volumes will not be created.
<details>
e.g 

```
$ ks get pvc
test-disk-0-6vxjb                        Pending                                                                        longhorn-image-kvsfj   29m
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    harvesterhci.io/imageId: default/image-kvsfj
    harvesterhci.io/owned-by: '[{"schema":"kubevirt.io.virtualmachine","refs":["default/test"]}]'
    volume.beta.kubernetes.io/storage-provisioner: driver.longhorn.io
  creationTimestamp: "2022-03-09T02:23:46Z"
  finalizers:
  - kubernetes.io/pvc-protection
  name: test-disk-0-6vxjb
  namespace: default
  resourceVersion: "4085568"
  uid: 70ee3e51-092a-4ae0-ac6e-f0b66c777b12
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: "9223372036854775807"
  storageClassName: longhorn-image-kvsfj
  volumeMode: Block
status:
  phase: Pending
```

</details>

Reply:
longhornio/longhorn-manager:v1.2.4-rc1

The ISO is built per https://github.com/harvester/harvester-installer/pull/240, 
![image](https://user-images.githubusercontent.com/31133476/157404835-6c537e5d-8b6b-49d3-a9f0-c80719d21d2a.png)

![image](https://user-images.githubusercontent.com/31133476/157405141-85dd7c8a-dd61-48ae-930d-8c7ba7b8997e.png)

kl longhorn-manager-qnr6q -o JSON  | grep image
            ..
                "image": `"longhornio/longhorn-manager:v1.2.4-rc1",`
                "imagePullPolicy": "IfNotPresent",
                "image": "docker.io/longhornio/longhorn-manager:v1.2.4-rc1",
                "imageID": "sha256:d5f302f87a2db67c8d0acdc4abdde37be5efb0b84cb47b0d171969c73dc72a02",

```
k get pvc
NAME       STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
..
vol-2t     Bound    pvc-b6b977d8-1f9f-443d-8f26-dfb774b75d23   2000Gi     RWX            longhorn       9h
vol-300g   Bound    pvc-95a08266-c865-4282-a8df-78028d608482   300Gi      RWX            longhorn       9h
```

Reply:
Update:
When creating Harvester Volume, if it is image-backed (Guangbo's case), it implicitly has 2 operations: create and write,  the over-sized volume will stuck in "Ponding" status;  if it is none-image related, the over-sized goes into "Bound" quickly.